.github/workflows/test.yml

```diff
@@ -0,0 +1,208 @@
+name: test
+on:
+  - push
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+jobs:
+  tools:
+    name: tools
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          elixir-version: 1.16.1
+          gleam-version: 1.0.0
+          otp-version: "26"
+      - uses: actions/setup-python@v5
+        with:
+          cache: pip
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "22"
+      - uses: jiro4989/setup-nim-action@v2
+      - uses: goto-bus-stop/setup-zig@v2
+      - uses: uncenter/setup-taplo@v1
+      - uses: actions/setup-go@v5
+        with:
+          go-version: stable
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.3"
+      - uses: dart-lang/setup-dart@v1
+      - uses: crystal-lang/install-crystal@v1
+      - uses: hasnep/setup-roc@v0.4.0
+      - uses: denoland/setup-deno@v2
+        with:
+          deno-version: v1.x
+      - uses: taiki-e/install-action@just
+      - uses: ocaml/setup-ocaml@v3
+        with:
+          ocaml-compiler: "5.1"
+      - uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: "21"
+      - uses: haskell-actions/setup@v2
+        with:
+          cabal-version: latest
+          ghc-version: latest
+      - uses: julia-actions/cache@v2
+      - uses: cargo-bins/cargo-binstall@main
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - run: rustup toolchain install stable --profile minimal
+      - run: rustup component add rustfmt clippy
+      - name: pyink
+        run: ( ( ( which pipx && pipx install pyink ) ) || echo "Unable to install tool" )
+      - name: purs-tidy
+        run: ( ( ( which npm && npm i -g purs-tidy ) ) || echo "Unable to install tool" )
+      - name: mado
+        run: ( ( ( which brew && brew tap akiomik/mado https://github.com/akiomik/mado.git &&  brew install mado ) ) || echo "Unable to install tool" )
```

Comment:
There is the concept of Brewfile where you could configure all the things you need for the project via HomeBrew

I think it could be useful for you

https://homebrew-file.readthedocs.io/en/latest/usage.html

Reply:
So I have looked a bit more into Brewfiles, and I do not think it makes sense in this case. 

But [whalebrew](https://github.com/whalebrew/whalebrew), which homebrew-bundle uses to run Docker containers, could make a lot of sense to integrate. The lack of activity in the whalebrew repo does make a bit afraid though.